drivers/fuel_gauge/sbs_gauge/sbs_gauge.c

```diff
@@ -102,6 +138,18 @@ static int sbs_gauge_get_props(const struct device *dev, struct fuel_gauge_get_p
 	return ret;
 }
 
+static int sbs_gauge_set_props(const struct device *dev, struct fuel_gauge_set_property *props,
+			       size_t len)
+{
+	int ret = 0;
+
+	for (int i = 0; i < len; i++) {
+		ret |= sbs_gauge_set_prop(dev, props + i);
+	}
+
+	return ret;
```

Comment:
This will be garbage if we ever return 2 different errors from sbs_gauge_set_prop. Consider replacing with `return ret == 0 ? 0 : -EINVAL;`

Reply:
I was wondering about that, but I saw that the read function already worked that way, so I figured somebody liked it like that. I'm a bit curious about the nominal positive return values when some reads/writes failed, and others succeeded. I'm not sure what I'd do with that as the caller. How is it supposed to be used?

Reply:
Made https://github.com/zephyrproject-rtos/zephyr/pull/52733

Reply:
I still think this is weird, but #52992 has been opened to that end.

Reply:
Yeah I'd like to add the `__subsystem` bit then focus on https://github.com/zephyrproject-rtos/zephyr/issues/52992 to get finalized.